test/src/net/test_conn.c

```diff
@@ -37,16 +37,9 @@ test_conn(PG_FUNCTION_ARGS)
 	Assert(connection_connect(conn, host, NULL, port) >= 0);
 
 	/* should timeout */
-	PG_TRY();
-	{
-		connection_read(conn, response, 1);
-	}
-	PG_CATCH();
-	{
-		should_fail = true;
-	}
-	PG_END_TRY();
-	Assert(should_fail);
+	ret = connection_read(conn, response, 1);
```

Comment:
This read (and the one below) are expected to timeout. When running `test/sql/net.sql` on Windows, we would get a NOTICE (from Postgres..) because Postgres caught a 10060 Windows error that it couldn't parse. Incidentally, this is a timeout error and expected. But did you not see this in Windows when you ran the `net` test? We had to modify `test/sql/net.sql` to ignore NOTICE-level messages.

Reply:
The reason why an exception was thrown on timeout is because PG actually wraps the socket calls on Windows in functions that expect non-blocking sockets. Since it never expects blocking sockets, it doesn't expect a timeout either.

I fixed this for now by using the Winsock API directly on Windows, bypassing the PG wrapper. So, now it returns a proper timeout error.

However, I noticed that the corresponding SSL read does not return a regular timeout error. Instead it returns`SSL_ERROR_WANT_READ`. Since the test code here just captures any exception, we don't notice. So question is if `SSL_ERROR_WANT_READ` is a valid return code for a timeout in the underlying socket.

Reply:
So you mean the underlying socket doesn't get an error...?

Reply:
that is unclear.